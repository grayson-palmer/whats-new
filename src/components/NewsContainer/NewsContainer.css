.newsContainer {
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  height: 100vh;
  overflow: scroll;
}