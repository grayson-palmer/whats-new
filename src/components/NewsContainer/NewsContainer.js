import React from 'react';
import './NewsContainer.css'
import NewsArticle from '../NewsArticle/NewsArticle';

const NewsContainer = (props) => {
  const localNews = props.local.map(news => {
    return <NewsArticle 
      key={news.id}
      headline={news.headline}
      body={news.description}
      img={news.img}
      link={news.url}
    />
  })
  return (
    <section className='newsContainer'>
      {localNews}
    </section>
  )
}

export default NewsContainer;