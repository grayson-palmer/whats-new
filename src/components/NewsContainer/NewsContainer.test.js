import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';

describe('NewsArticle', () => {
  it('should be an instance of component', () => {
    const wrapper = shallow(<NewsContainer 
      newsType = {[{
        headline: 'Thanos Back Again', 
        description: 'Purp dude is back', 
        img: 'imgPath',
        url: 'urlPath',
        id: 1
      }]}
    />)
    expect(wrapper).toMatchSnapshot();
  })
})