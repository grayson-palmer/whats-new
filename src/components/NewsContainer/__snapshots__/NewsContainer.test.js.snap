// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should be an instance of component 1`] = `
<section
  className="newsContainer"
>
  <NewsArticle
    body="Purp dude is back"
    headline="Thanos Back Again"
    img="imgPath"
    key="1"
    link="urlPath"
  />
</section>
`;
